### ISSUE TITLE
Furnace recipes causing issues with Slimefun4

### ISSUE THREAD

**Issue by LcyDev** [WolfyScript/CustomCrafting#221]
### What happened?

When you use an `Enhanced Furnace` of [Slimefun4](https://github.com/Slimefun/Slimefun4) and try to use recipes made in Custom Crafting, everything's fine until the last item gets smelted.

![image](https://user-images.githubusercontent.com/72966048/214649924-ce8c5743-9a82-4abf-acfe-8d1e1010bfe2.png)

Once that happens, an error will pop up in the console.

### How can the bug be reproduced?

1. Install Slimefun4.
2. Make a CC furnace recipe.
3. `/sf give <player> ENHANCED_FURNACE`
4. Try smelting the custom recipe in it.
5. Check the console for errors.

### Expected Behaviour?

This shouldn't happen

### Plugin version

CustomCrafting-4.16.5.1 - WolfyUtils-4.16.9.6

### Server software

Purpur (1.19.3) b1900

### Relevant log output

```shell
https://paste.gg/p/ConaII_/ba285dbaa2654c33a3544a0e8b714ba9

https://paste.gg/p/ConaII_/ce0b4a7588a34a96a7b233f099e54ec7
```

### Submit issue

- [X] read the Wiki and especially the FaQ, and made sure there is no fix to your issue,
- [X] searched for and ensured there isn't already an open issue regarding this,
- [X] are running the latest version of your server software (Spigot, Paper, etc.), and plugins (WolfyUtilities, CustomCrafting).

**Comment by LcyDev** [comment]
https://github.com/Slimefun/Slimefun4/issues/3741

**Comment by J3fftw1** [comment]
This issue is fixed in https://github.com/Slimefun/Slimefun4/pull/3756

**Comment by WolfyScript** [comment]
> This issue is fixed in [Slimefun/Slimefun4#3756](https://github.com/Slimefun/Slimefun4/pull/3756) For some reason CustomCrafting adds a null item. We still should have checked for it in the first place but still weird a null item is being added.

Yeah, I don't know why it happens, I guess it can be null by desing (at least that's what the API says). 
I had to also check if the smelting item is null, because another plugin caused the same issue before.   
https://github.com/WolfyScript/CustomCrafting/blob/73bebaa50537ac36ff0bd492375eb067d9014242/src/main/java/me/wolfyscript/customcrafting/utils/cooking/SmeltAPIAdapter.java#L92

**Comment by J3fftw1** [comment]
im guessing your check happens later then we check the smelting for some reason.
Might also just be spigot jank
